e2e/e2e-script.sh

```diff
@@ -49,17 +50,30 @@ az vmss run-command invoke \
 
 declare -a files=("apiserver.crt" "ca.crt" "client.key")
 for file in "${files[@]}"; do
-    content=$(az vmss run-command invoke \
+    for i in $(seq 1 10); do
+        set +e
+        content=$(az vmss run-command invoke \
                 -n $VMSS_NAME \
                 -g $MC_RESOURCE_GROUP_NAME \
                 --command-id RunShellScript \
                 --instance-id 0 \
-                --scripts "cat /etc/kubernetes/certs/$file | base64 -w 0" | \
+                --scripts "cat /etc/kubernetes/certs/$file | base64 -w 0" \
+                -ojson | \
                 jq -r '.value[].message' | \
                 awk '/stdout/{flag=1;next}/stderr/{flag=0}flag' | \
                 awk NF \
-            )
-    addJsonToFile $file $content
+        )
+        retval=$?
+        set -e
+        if [ "$retval" -ne 0 ]; then
+            echo "retrying attempt $i"
+            sleep 10s
+            continue
+        fi
+        break;
+    done
+    [ "$retval" -eq 0 ]
```

Comment:
what's the intent here? validate the final retval and exit if not zero? not super accustomed to this syntax without `if; then; fi`

Reply:
It's basically a shorthand for the following (since we have `set -e`):

```bash
if [ "$retval" -eq 0 ]; then
    exit 1
fi
```

I can refactor to echo something and exit explicitly if that makes more sense.

Reply:
fine as is, just wanted to double check 👍 learned something new today